app/components/layout/install-pwa-prompt-modal.tsx

```diff
@@ -0,0 +1,71 @@
+import { useLoaderData, useFetcher } from "@remix-run/react";
+import { AnimatePresence } from "framer-motion";
+
+import type { loader } from "~/routes/_layout+/_layout";
+import { usePwaManager } from "~/utils/pwa-manager";
+import { Button } from "../shared/button";
+import { useRef } from "react";
+
+export function InstallPwaPromptModal() {
+  const { hideInstallPwaPrompt } = useLoaderData<typeof loader>();
+  const fetcher = useFetcher();
+  let optimisticHideInstallPwaPrompt = hideInstallPwaPrompt;
+  if (fetcher.formData) {
+    optimisticHideInstallPwaPrompt =
+      fetcher.formData.get("pwaPromptVisibility") === "hidden";
+  }
+  const hidePwaPromptForm = useRef<HTMLFormElement | null>(null);
+
+  const { promptInstall } = usePwaManager();
+  return optimisticHideInstallPwaPrompt ? null : (
+    <AnimatePresence>
```

Comment:
I think just like this its not doing anything. You need to add the motion.div inside it. We have some examples. Would be nice to have it animated.